ckanext/googleanalytics/model.py

```diff
@@ -22,41 +22,108 @@ class PackageStats(Base):
     package_id = Column(types.UnicodeText, nullable=False, index=True, primary_key=True)
     visit_date = Column(types.DateTime, default=datetime.now, primary_key=True)
     visits = Column(types.Integer)
+    entrances = Column(types.Integer)
+    downloads = Column(types.Integer)
 
     @classmethod
     def get(cls, id):
         return model.Session.query(cls).filter(cls.package_id == id).first()
 
     @classmethod
-    def update_visits(cls, item_id, visit_date, visits):
+    def update_visits(cls, item_id, visit_date, visits=0, entrances=0, downloads=0):
         '''
         Updates the number of visits for a certain package_id
         or creates a new one if it is the first visit for a certain date
 
         :param item_id: package_id
         :param visit_date: visit date to be updated
-        :param visits: number of visits until visit_date
+        :param visits: number of visits during date
+        :param entrances: number of entrances during date
         :return: True for a successful update, otherwise False
         '''
         package = model.Session.query(cls).filter(cls.package_id == item_id).filter(cls.visit_date == visit_date).first()
         if package is None:
-            package = PackageStats(package_id=item_id, visit_date=visit_date, visits=visits)
+            package = PackageStats(package_id=item_id, visit_date=visit_date, visits=visits, entrances=entrances, downloads=downloads)
             model.Session.add(package)
-        else:
+        elif visits != 0:
             package.visits = visits
+        elif entrances != 0:
+            package.entrances = entrances
```

Comment:
What if both visits and entrances are different from 0? Should they both have their own if-statement?

Reply:
yeah, you are correct.